src/detect-engine-dns.c

```diff
@@ -119,6 +120,48 @@ int DetectEngineInspectDnsQueryName(ThreadVars *tv,
     return r;
 }
 
+int DetectEngineInspectDnsResponse(ThreadVars *tv,
```

Comment:
we had a DetectEngineInspectDnsResponse, it wrapped around a 'generic list' inspect function. Now it's a buffer inspect function. This means that the old logic is gone which means that Lua scripts will be broken.

detect-lua.c:1030 sets up lua scripts in the 'generic list' for the list with name 'dns_response'. Luckily this name is only used internally, so you can change it. But you will have to make sure not to break this.

Reply:
A possible source of confusion is that internally we name 'buffers'. These names may be the same as a keyword inspecting them, but this doesn't have to be the case. It would probably be good to first rename the existing 'dns_response' buffer to 'dns_response_generic' or something. Then in a 2nd commit, add your 'dns_response' keyword. That should solve the confusion around what 'dns_response' means.